pkg/providers/amifamily/ami.go

```diff
@@ -160,8 +160,30 @@ func (p *Provider) getDefaultAMIFromSSM(ctx context.Context, nodeTemplate *v1alp
 		if err != nil {
 			return nil, err
 		}
-		amis = append(amis, AMI{Name: ssmOutput.Name, AmiID: amiID, Requirements: ssmOutput.Requirements})
+		amis = append(amis, AMI{AmiID: amiID, Requirements: ssmOutput.Requirements})
 	}
+	// Creating selector filter by making a string of amiIds into a comma delineated string
```

Comment:
I think you could move L165 through L186 to a helper function called something like `fillAMINamesForSSMAMIs`. Maybe you have a better name :D